recipe_app/views.py

```diff
@@ -1,20 +1,123 @@
+
+
+@ login_required
+def add_recipe(request):
+    if request.method == 'POST':
+        form = AddRecipeForm(request.POST)
+        if form.is_valid():
+            data = form.cleaned_data
+            new_recipe = Recipe.objects.create(
+                title=data['title'],
+                description=data['description'],
+                time_required=data['time_required'],
+                instructions=data['instructions'],
+                author=data['author']
+            )
+            return HttpResponseRedirect(reverse('homepage'))
+
+    form = AddRecipeForm()
+    return render(request, 'generic_form.html', {'form': form})
+
+
+@ staff_member_required
+def add_author(request):
+    if request.user.is_staff:
+        if request.method == 'POST':
+            form = AddAuthorForm(request.POST)
+            if form.is_valid():
+                data = form.cleaned_data
+                myuser = User.objects.create_user(
+                    username=data['username'], password=data['password'])
+                Author.objects.create(
+                    name=data['name'], bio=data['bio'], user=myuser)
+                return HttpResponseRedirect(reverse('homepage'))
+
+        form = AddAuthorForm()
+        return render(request, 'generic_form.html', {'form': form})
+    else:
+        raise Http404
+
+
+def login_view(request):
+    if request.method == 'POST':
+        form = LoginForm(request.POST)
+        if form.is_valid():
+            data = form.cleaned_data
+            user = authenticate(
+                request, username=data['username'], password=data['password'])
+            if user:
+                login(request, user)
+                return HttpResponseRedirect(request.GET.get('next', reverse('homepage')))
+
+    form = LoginForm()
+    return render(request, 'generic_form.html', {'form': form})
+
+
+def logout_view(request):
+    logout(request)
+    return HttpResponseRedirect(reverse('homepage'))
+
+
+@ login_required
+def edit_recipe(request, recipe_id: int):
+    recipe = Recipe.objects.get(id=recipe_id)
+
+    if request.method == 'POST':
+        form = AddRecipeForm(request.POST)
+        if form.is_valid:
+            data = form.cleaned_data
+            recipe.title = data['title']
+            recipe.description = data['description']
+            recipe.time_required = data['time_required']
+            recipe.instructions = data['instructions']
+            recipe.author = data['author']
+            recipe.save()
+            return HttpResponseRedirect(reverse('recipe_detail', args=(recipe_id,)))
+
+    form = AddRecipeForm(initial={
+        'title': recipe.title,
+        'description': recipe.description,
+        'time_required': recipe.time_required,
+        'instructions': recipe.instructions,
+        'author': recipe.author
+    })
+    return render(request, 'generic_form.html', {'form': form})
```

Comment:
We need to "protect" this form with a conditional statement that checks to see if the logged in user is 1. a superuser OR 2. the author of the recipe they want to edit.

What is a conditional statement? What variable represents the logged in user? How can we compare the logged in user to another variable to see if they are a superuser or this recipe's author?